src/main/java/com/commercetools/pingpong/service/impl/GameServiceImpl.java

```diff
@@ -7,29 +7,29 @@
 import org.springframework.stereotype.Component;
 
 @Component
-public class GameServiceImpl implements GameService{
+public class GameServiceImpl implements GameService {
     private Game match;
 
     public GameServiceImpl() {
-        this.match = new Game(new Player("HeshamOne"), new Player("HeshamTwo"));
+        match = new Game(new Player("HeshamOne"), new Player("HeshamTwo"));
     }
     // FIXME: These hardcoded values will be removed later
 
     @Override
-    public void updateScore(final Message message) {
+    public void updateScore(Message message) {
         Player actingPlayer = getPlayer(message);
         if (match.areBothSetScoresZero() && !match.isSomeoneServing()) {
             match.setFirstServe(actingPlayer);
         } else {
             match.updateScoreOfPlayer(actingPlayer, message.getActionType());
             match.updateMatchScoreOfPlayers();
-            match.isGameOver();
```

Comment:
Actually I meant to do the `if (match.isMatchOver())` check here 😅

Like this it seems that you will always end the game, whenever you click the button.

Reply:
oh ok 👍 
0f7ce81de1de624edee1ff4102681bc06ac727cb